internal/cluster/transition_test.go

```diff
@@ -4693,6 +4693,122 @@ var _ = Describe("ValidationResult sort", func() {
 	})
 })
 
+var _ = Describe("finalizing timeouts", func() {
+	var (
+		ctx              = context.Background()
+		db               *gorm.DB
+		clusterId        strfmt.UUID
+		clusterApi       *Manager
+		mockEvents       *eventsapi.MockHandler
+		mockHostAPI      *host.MockAPI
+		mockMetric       *metrics.MockAPI
+		ctrl             *gomock.Controller
+		dbName           string
+		operatorsManager *operators.Manager
+		mockS3Api        *s3wrapper.MockAPI
+	)
+
+	BeforeEach(func() {
+		db, dbName = common.PrepareTestDB()
+		ctrl = gomock.NewController(GinkgoT())
+		mockEvents = eventsapi.NewMockHandler(ctrl)
+		mockHostAPI = host.NewMockAPI(ctrl)
+		mockMetric = metrics.NewMockAPI(ctrl)
+		mockS3Api = s3wrapper.NewMockAPI(ctrl)
+		operatorsManager = operators.NewManager(common.GetTestLog(), nil, operators.Options{}, nil, nil)
+		clusterId = strfmt.UUID(uuid.New().String())
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+		common.DeleteTestDB(db, dbName)
+	})
+	createCluster := func(status string, stage models.FinalizingStage, statusTimestamp, stageTimestamp time.Time) *common.Cluster {
+		cls := &common.Cluster{
+			Cluster: models.Cluster{
+				ID:               &clusterId,
+				Status:           swag.String(status),
+				StatusInfo:       swag.String(statusInfoFinalizing),
+				StatusUpdatedAt:  strfmt.DateTime(statusTimestamp),
+				OpenshiftVersion: "4.15",
+				EmailDomain:      "redhat.com",
+				Progress: &models.ClusterProgressInfo{
+					FinalizingStage:          stage,
+					FinalizingStageStartedAt: strfmt.DateTime(stageTimestamp),
+				},
+			},
+		}
+		Expect(db.Create(cls).Error).ToNot(HaveOccurred())
+		return cls
+	}
+	Context("soft timeouts disabled", func() {
+		BeforeEach(func() {
+			clusterApi = NewManager(getDefaultConfig(), common.GetTestLog().WithField("pkg", "cluster-monitor"), db, testing.GetDummyNotificationStream(ctrl),
+				mockEvents, nil, mockHostAPI, mockMetric, nil, nil, operatorsManager, nil, mockS3Api, nil, nil, nil, false)
+		})
+	})
```

Comment:
This looks like it doesn't do anything. Please remove it.

Reply:
OK